### PR TITLE
Require only app-extension-safe APIs

### PR DESCRIPTION
**Pull request by mattlarose**
tl;dr: Marks Rive library as safe, to prevent potential App Store rejection.

If an app extension (e.g., a widget) depends on Rive, when you build your app you'll be warned that you're `Linking against a dylib which is not safe for use in application extensions`. Apple may reject the app for this. 

A library is safe for application extensions if `APPLICATION_EXTENSION_API_ONLY = YES`.

Right now, `APPLICATION_EXTENSION_API_ONLY` is unspecified in Rive, but Rive is only using APIs that are safe for app extensions. Since Rive _is_ safe, I marked it as safe. If an unsafe API is ever used in the future, Xcode will generate errors. I'd guess that given [the usual unsafe APIs](https://developer.apple.com/library/archive/documentation/General/Conceptual/ExtensibilityPG/ExtensionOverview.html#//apple_ref/doc/uid/TP40014214-CH2-SW6), this seems unlikely ever to be a problem for Rive.

I didn't see some project file management setup, so I edited the `.pbxproj` directly.

### CODE REVIEW

**Comment by mattlarose** [comment]
silly fork branch mishap, reopened in #267
